client/index.js

```diff
@@ -59,10 +59,18 @@ var onSocketMsg = {
 };
 
 var newConnection = function() {
+	var hostname = urlParts.hostname;
+
+	if (urlParts.hostname === '0.0.0.0') {
+		if (window.location.hostname && !!~window.location.protocol.indexOf('http')) {
```

Comment:
Could you add that to the code as a comment though? This is not immediately clear, and since there are no tests yet, it's easy for this to get accidentally removed.